pkg/api/types.go

```diff
@@ -221,6 +224,17 @@ type GCEPersistentDisk struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// A ISCSI Disk can only be mounted as read/write once.
+type ISCSIDisk struct {
```

Comment:
@thockin I understand and agree with the idea that users defining their own pets in VolumeSource will eventually go away (replacing the network volumes with a struct like you suggested in #4055).

This plugin and @calfonso NFSMount plugin are orthogonal issues that can be developed independently of what I'm doing, but then refactored into whatever design comes next.  I have no problem doing that refactoring work as a follow-on task.